woocommerce/woocommerce-android
# Integrate `ProductSelectorFragment` into Order creation

#### Order creation - products multi-selection
Closes: https://github.com/woocommerce/woocommerce-android/issues/8420

### Description
* Integrated `ProductSelectorFragment` into Order creation - product/variation multi-selection flow. The old solution allowed the selection of a single product. The new one provides a multi-selection of items.
* Deleted old product and variation picker code
* Removed part of Order creation UI test related to the old, deleted product selection code.

### Testing instructions
Go to the Orders screen and create a new Order. Tap + Add product screen and verify the multi-selection adds products/variations correctly to the order. Important scenarios to test:

* Can select multiple products and variations at the same time
* Products and variations already added to the order are marked as selected when returning to the product picker page
* Unselecting previously selected products/variations should remove them from the order
* In case the selected product quantity was updated on the order page it should be preserved when opening the product picker and returning back to the order
* Tapping on the selected product on the Orders page should redirect to the Item Details screen and allow removing the product
* The coupon edition screen (product pickers) should remain working without any changes as well.

### Images/gif


https://user-images.githubusercontent.com/4527432/223449344-2b167912-ee09-4ece-adf8-7aaee2b2c5c6.mp4




We'll do it when merging a final PR to trunk.
